.github/workflows/cd-dgraph.yml

```diff
@@ -22,6 +22,19 @@ jobs:
           go mod tidy
           make regenerate
           git diff --exit-code -- .
+      - name: Set Badger Release Version
+        run: |
+          #!/bin/bash
+          BADGER_RELEASE_VERSION=$(cat go.mod | grep -i badger | awk '{print $2}')
```

Comment:
Can use `go list -f '{{.Version}}' -m github.com/dgraph-io/badger/v3`

Reply:
I was thinking about this ^, I think we really need to have a talk here ^. I did not like pinning to `github.com/dgraph-io/badger/v3` where we specifically choose v3. We wanted to make semver changes to badger repo, and possibly cut 4.0.0. This may complicate that process.  

Also, when I tried `github.com/dgraph-io/badger` <- it spat out a v 1.x* version.

Reply:
Addressing the latter point: Go considers `github.com/dgraph-io/badger` and `github.com/dgraph-io/badger/v3` to be two totally distinct modules (and hence they have two distinct import paths).  The expectation is that whenever a major version is released, dependent projects must modify the import path of the dependency to point to the new module (so it is not simply an upgrade of an existing module).

I do see your point - making a major version upgrade on Badger would require remembering to modify the Makefile, which is not great.

Reply:
yeah +1 ^ same issues noticed, lets talk and make a decision today. The nice part is we have 2 working approaches. A choice is to be made.